docs/DeveloperGuide.adoc

```diff
@@ -153,6 +163,101 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+=== List feature
+
+The List mechanism is facilitated by the combination of three classes `MainWindow`, `Model` and `Context`.
+
+`Context` is an immutable utility class that describes the nature of the current view of the app. It stores state information about the `ContextType` as well as an optional `Object` of interest. A `Context` may be instantiated either by
+
+* one of the factory methods `newListActivityContext` or `newListContactContext`
+* invoking the constructor with either an `Activity` or `Contact` of interest
+
+The `Context` class exposes relevant getter methods `Context#getType`, `Context#getActivity` and `Context#getContact`, and thus supports contextual behaviour of other commands.
+
+During execution, the List command creates the intended `Context` and updates the `Model` with it, along with the appropriate `FilteredList`. As `MainWindow` tracks both `FilteredList`, JavaFX will automatically re-render the contained card entries when the respective list undergoes structural changes.
+
+The updated `ContextType` of the `Model` is then returned to the `MainWindow` UI controller via the `CommandResult`, where it is used to switch the content displayed.
+
+Given below is an example usage scenario of a user intending to view the list of activities.
+
+Step 1. The user launches the `SplitWiser` application for the first time, which defaults to displaying the list of contacts.
+
+Step 2. The user executes the command string `list a/` to prompt the app to display the full list of activities.
+
```

Comment:
This could list out the remaining steps to be more comprehensive (e.g. how the logic manager selects the command parser, how the command parser parses the command string into a command, how the command is executed, etc.).